[export] Prevent specialization on backends

angelayi opened the pull request:
Summary: https://github.com/pytorch/pytorch/issues/118289 shows that sometimes we will decompose into backend-specific operators, causing some specializations. We should probably avoid this by disabling these by default?

Test Plan: CI

Differential Revision: D53241300

zhxchen17 commented on torch/export/_trace.py:
```diff
@@ -70,6 +70,17 @@ class ExportDynamoConfig:
 
 
+@contextmanager
+def ignore_backend_decomps():
```
extra cautious lol
```suggestion
def _ignore_backend_decomps():
```

angelayi replied:
ooh yes ty

angelayi committed:
[export] Prevent specialization on backends (#118683)

Summary:

https://github.com/pytorch/pytorch/issues/118289 shows that sometimes we will decompose into backend-specific operators, causing some specializations. We should probably avoid this by disabling these by default?

Test Plan: CI

Reviewed By: zhxchen17

Differential Revision: D53241300
